File: src/main/java/com/peaqockback/peaqock/entity/Article.java
package com.peaqockback.peaqock.entity;

import java.util.Date;
//import java.text.SimpleDateFormat;

public class Article {
    private int id;
    private String titre;
    private String contenu;
    private String auteur;
    private String urlImage;
    private Date dateCreation;

    public Article() {
    }

    public Article(int id, String titre, String contenu, String auteur, String urlImage) {
        this.id = id;
        this.titre = titre;
        this.contenu = contenu;
        this.auteur = auteur;
        this.urlImage = urlImage;
        this.dateCreation = new Date();
    }

    public Article(int id, String titre, String contenu, String auteur, String urlImage, Date dateCreation) {
        this.id = id;
        this.titre = titre;
        this.contenu = contenu;
        this.auteur = auteur;
        this.urlImage = urlImage;
        this.dateCreation = dateCreation;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitre() {
        return titre;
    }

    public void setTitre(String titre) {
        this.titre = titre;
    }

    public String getContenu() {
        return contenu;
    }

    public void setContenu(String contenu) {
        this.contenu = contenu;
    }

    public String getAuteur() {
        return auteur;
    }

    public void setAuteur(String auteur) {
        this.auteur = auteur;
    }

    public String getUrlImage() {
        return urlImage;
    }

    public void setUrlImage(String urlImage) {
        this.urlImage = urlImage;
    }

    public Date getDateCreation() {
        return dateCreation;
    }

    public void setDateCreation(Date dateCreation) {
        this.dateCreation = dateCreation;
    }
}

File: src/app/article/article.ts
export class Article{

  private _id: number;
  private _titre: string;
  private _contenu: string;
  private _auteur: string;
  private _urlImage: string;
  private _dateCreation: string;


  constructor(id?: number, titre?: string, contenu?: string, auteur?: string,
              urlImage?: string)
  {
    this._id = id;
    this._titre = titre;
    this._contenu = contenu;
    this._auteur = auteur;
    this._urlImage = urlImage;
    var dateCreation = new Date();
    this._dateCreation = dateCreation.getFullYear() + '-' + (dateCreation.getMonth() + 1) + '-' + dateCreation.getDate();
  }


  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get titre(): string {
    return this._titre;
  }

  set titre(value: string) {
    this._titre = value;
  }

  get contenu(): string {
    return this._contenu;
  }

  set contenu(value: string) {
    this._contenu = value;
  }

  get auteur(): string {
    return this._auteur;
  }

  set auteur(value: string) {
    this._auteur = value;
  }

  get urlImage(): string {
    return this._urlImage;
  }

  set urlImage(value: string) {
    this._urlImage = value;
  }

  get dateCreation(): string {
    return this._dateCreation;
  }

  set dateCreation(value: string) {
    this._dateCreation = value;
  }


}
